### PR TITLE
Implement git.Cryptoer

### PR DESCRIPTION
**Pull request by patrickxb**
cc @oconnor663

### CODE REVIEW

File: go/git/crypto.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+// Box encrypts the plaintext with the most current key for the given team. It yields a NaCl
+// ciphertext and nonce, and also says which generation of the key it used.
+func (c *Crypto) Box(ctx context.Context, plaintext []byte, teamSpec keybase1.TeamIDWithVisibility) (*keybase1.EncryptedGitMetadata, error) {
+	team, err := c.loadTeam(ctx, teamSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	// For v1 of this, we'll use KBFS key so that we don't need to backfill in
```
Comment (maxtaco) [inline go/git/crypto.go]: as discussed, let's fix this now...

File: go/git/interfaces.go [hunk 2]
```diff
@@ -16,8 +16,8 @@ type Teamer interface {
 type Cryptoer interface {
 	// Box encrypts the plaintext with the most current key for the given team. It yields a NaCl
 	// ciphertext and nonce, and also says which generation of the key it used.
-	Box(ctx context.Context, plaintext []byte, team keybase1.TeamIDWithVisibility) (ciphertext []byte, nonce keybase1.BoxNonce, keyGeneration keybase1.PerTeamKeyGeneration, err error)
+	Box(ctx context.Context, plaintext []byte, team keybase1.TeamIDWithVisibility) (*keybase1.EncryptedGitMetadata, error)
 	// Unbox decrypts the given ciphertext with the given nonce, for the given generation of the
 	// given team. Can return an error. Will return a non-nil plaintext on success.
-	Unbox(ctx context.Context, ciphertext []byte, nonce keybase1.BoxNonce, team keybase1.TeamIDWithVisibility, keyGeneration keybase1.PerTeamKeyGeneration) (plaintext []byte, err error)
+	Unbox(ctx context.Context, team keybase1.TeamIDWithVisibility, metadata *keybase1.EncryptedGitMetadata) (plaintext []byte, err error)
```
Comment (maxtaco) [inline go/git/interfaces.go]: why a pointer here?

Comment (patrickxb) [reply go/git/interfaces.go]: Just to match what is returned by Box

File: go/git/crypto.go (hunk 1 above)
Comment (patrickxb) [reply go/git/crypto.go]: ok

**Comment by maxtaco** [comment]
Can hopefully work on it tonight.

**Comment by maxtaco** [comment]
Bust the teams cache [here](https://github.com/keybase/client/blob/master/go/teams/storage.go#L82-L83)

**Comment by maxtaco** [comment]
Keybase server pull here: https://github.com/keybase/keybase/pull/1661

**Comment by maxtaco** [comment]
OK, merged and pushed live.

**Comment by patrickxb** [comment]
Thanks, I'll make changes in the morning to use git key, bust cache.

File: go/git/crypto_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	}
+	plaintext, err := libkb.RandBytes(80)
+	require.NoError(tc.T, err)
+
+	c := NewCrypto(tc.G)
+	boxed, err := c.Box(context.Background(), plaintext, teamSpec)
+	require.NoError(tc.T, err)
+	require.NotNil(tc.T, boxed)
+	require.EqualValues(tc.T, boxed.Gen, 1)
+	require.Len(tc.T, boxed.N, libkb.NaclDHNonceSize)
```
Comment (oconnor663) [inline go/git/crypto_test.go]: Could we also check that the nonce isn't all zeroes here, just for kicks?

**Comment by keybase-ci-visdiff** [comment]
The commits 3e160f2e6af7...f553710578c2 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504788984390-3e160f2e6af7-f553710578c2-linux/Markdown-emoji-3e160f2e6af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504788984390-3e160f2e6af7-f553710578c2-linux/Markdown-emoji-f553710578c2.png)  
   Markdown-emoji comparison
</summary>

File: go/git/crypto_test.go (hunk 3 above)
Comment (maxtaco) [reply go/git/crypto_test.go]: 👍

**Comment by patrickxb** [comment]
Ok, no complaining on the day that libkb.RandBytes() returns all zeros and the test fails 😄

**Comment by oconnor663** [comment]
I promise not to blame it on you if that happens :-D

**Comment by patrickxb** [comment]
*when* that happens ...

**Pull request merged**